### PR TITLE
Eventing TLS: update destination and Kref types

### LINKED ISSUE #2712: Eventing TLS: Update Destination and KReference
As the [Eventing TLS feature track](https://docs.google.com/document/d/1H-x_oji8LqkCyd7tlsSyclmUe7FAmEJPgRxOU_0pkn8/edit?resourcekey=0-lzDIPJsZOP3G17QE_g1lHw#) describes we should update Destination and KReference types as follow:

```golang
// Destination represents a target of an invocation over HTTP(S).
type Destination struct {
	// Ref points to an Addressable.
	// +optional
	Ref *KReference `json:"ref,omitempty"`

	// URI can be an absolute URL(non-empty scheme and non-empty host) pointing to the target or a relative URI. Relative URIs will be resolved using the base URI retrieved from Ref.
	// +optional
	URI *apis.URL `json:"uri,omitempty"`

	// Certification Authority (CA) certificates in PEM format
        // according to https://www.rfc-editor.org/rfc/rfc7468.
        // If set, these CAs are appended to the set of CAs provided
        // by the Addressable target, if any.
        // +optional
	CACerts *string `json:”CACerts,omitempty”`
}

// KReference contains enough information to refer to another object.
// It's a trimmed down version of corev1.ObjectReference.
type KReference struct {
	// …

	// Address points to a specific Address Name.
	// +optional
	Address *string `json:"address,omitempty"`
}
```

## Additional Info

- https://docs.google.com/document/d/1H-x_oji8LqkCyd7tlsSyclmUe7FAmEJPgRxOU_0pkn8/edit?resourcekey=0-lzDIPJsZOP3G17QE_g1lHw#

/good-first-issue
/kind feature
/area API

### PR DESCRIPTION
**Pull request by rahulii**
# Changes

- :broom: This PR updates Destination and Kreference types

/kind <kind>

Fixes #2712 

**Release Note**

```release-note

```

**Docs**

```docs
https://docs.google.com/document/d/1H-x_oji8LqkCyd7tlsSyclmUe7FAmEJPgRxOU_0pkn8/edit?resourcekey=0-lzDIPJsZOP3G17QE_g1lHw#heading=h.lu74k6uah3r
```

### CODE REVIEW

**Comment by rahulii** [comment]
hey @pierDipi , should we also update `Destination` in `v1beta1` ?

**Comment by pierDipi** [comment]
> hey @pierDipi , should we also update `Destination` in `v1beta1` ?

Yes

File: apis/duck/v1/destination.go [hunk 1]
```diff
@@ -31,6 +31,13 @@ type Destination struct {
 	// URI can be an absolute URL(non-empty scheme and non-empty host) pointing to the target or a relative URI. Relative URIs will be resolved using the base URI retrieved from Ref.
 	// +optional
 	URI *apis.URL `json:"uri,omitempty"`
+
+	// Certification Authority (CA) certificates in PEM format
```
Comment (pierDipi) [inline apis/duck/v1/destination.go:35]:
```suggestion
	// CACerts are Certification Authority (CA) certificates in PEM format
```

**Comment by rahulii** [comment]
@pierDipi PTAL!

**Review by pierDipi — APPROVED** [review]
Thanks @rahulii 

/lgtm
/approve

**Pull request merged**